engine/src/main/java/org/camunda/bpm/engine/impl/form/type/DateFormType.java

```diff
@@ -94,6 +98,12 @@ public Object convertFormValueToModelValue(Object propertyValue) {
     if (propertyValue==null || "".equals(propertyValue)) {
       return null;
     }
+    if (propertyValue instanceof Date) {
+      return propertyValue;
+    }
```

Comment:
If I understood you correctly, you propose only to keep these lines ⬆️ 

The method `convertFormValueToModelValue` was deprecated in favor of `convertToModelValue`. Could you explain why you want to provide a fix to a deprecated method?